### ISSUE TITLE
[bug] TFS 1.3 House invite list doesn't parse guild ranks

### ISSUE THREAD

**Issue by jerryb1988** [otland/forgottenserver#2240]
### Steps to reproduce (include any configuration/script required to reproduce)
1. Add RankName@GuildName to any house list (guest, subowner, door)

### Expected behaviour
Only players that have a rank name == RankName should be able to enter.

Reference for additional functionality:
 http://tibia.wikia.com/wiki/House#The_House_Spells

### Actual behaviour
Anybody in the guild can enter, regardless of rank.

### Environment
house.cpp -> line 420

Adds the entire guild instead of only guild members with mentioned rank.

```
void AccessList::parseList(const std::string& list)
{
	playerList.clear();
	guildList.clear();
	expressionList.clear();
	regExList.clear();
	this->list = list;
	if (list.empty()) {
		return;
	}

	std::istringstream listStream(list);
	std::string line;

	while (getline(listStream, line)) {
		trimString(line);
		trim_left(line, '\t');
		trim_right(line, '\t');
		trimString(line);

		if (line.empty() || line.front() == '#' || line.length() > 100) {
			continue;
		}

		toLowerCaseString(line);

		std::string::size_type at_pos = line.find("@");
		if (at_pos != std::string::npos) {
			addGuild(line.substr(at_pos + 1));
		} else if (line.find("!") != std::string::npos || line.find("*") != std::string::npos || line.find("?") != std::string::npos) {
			addExpression(line);
		} else {
			addPlayer(line);
		}
	}
}
```
### Example

Player1 is a Rank1 of Guild1.
Player2 is a Rank2 of Guild1.

Add Rank1@Guild1 to invite list.
Rank1 and Rank2 can enter.

Only Player1 should be allowed to enter. Player2 should receive "Not invited."

**Comment by Codinablack** [comment]
I can confirm this bug! According to the reference on tibia.wiki that is indeed the intended functionality of the @ symbol. Used 2 ways, but only coded to work 1 way. 

I have tested it, it will not work by rank.

I have also looked, I am unsure on what would be the best way to fix this issue. I know tapping in after the check for the @ symbol is where I need to be, but there are options. Should I construct a player reference and convert the string to check if the players rank matches the text? How can I do that without the player name. Even then I didn't see a method for that, only getting rank by id or level. I wasn't really sure how all this works, if someone could point me into the right direction.

I also think that it should be a loop that checks, maybe by getting all the members of that guild to run the check against?
